src/App.js

```diff
@@ -1,17 +1,110 @@
 import React from 'react';
 import './App.css';
+import ClassNames from 'classnames';
+import { TodoList } from './componenets/TodoList';
 
-import users from './api/users';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-function App() {
-  return (
+class App extends React.Component {
+  state = {
+    todos: todosFromServer.map(todo => ({
+      ...todo,
+      user: usersFromServer.find(user => user.id === todo.userId),
+    })),
+    isHidden: true,
+    errorMessage: '',
+  }
+
+  timeToShowError = false;
+
+  validation = (event) => {
+    const form = event.target.closest('#formAdder');
+
+    if (!form.title.value && this.timeToShowError) {
+      this.setState({
+        isHidden: false,
+        errorMessage: 'enter title',
+      });
+
+      return false;
+    }
+
+    if (!form.userSelect.value && this.timeToShowError) {
+      this.setState({
+        isHidden: false,
+        errorMessage: 'chose user',
+      });
+
+      return false;
+    }
+
+    this.setState({
+      isHidden: true,
+      errorMessage: '',
+    });
+
+    return true;
+  }
+
+  Add = (event) => {
+    event.preventDefault();
+
+    this.timeToShowError = true;
+
+    if (!this.validation(event)) {
+      return;
+    }
+
+    const newTodo = {
+      id: [...this.state.todos].sort((a, b) => b.id - a.id)[0].id + 1,
```

Comment:
it's too complicated solution for setting a new id, sorting is a heavy process

Reply:
but it garanted that Id will unical

Reply:
use uuid library, this would be a better solution

Reply:
what if your array is of 1000 items?..